### DISCUSSION TITLE
Rules as list

### DISCUSSION THREAD

**Discussion by Nik3181** [zeroSteiner/rule-engine#24] [Q&A] [answered]
Is it possible to create a list containing rule definition, then use matches method that return rules that matches the object ?

**Comment by zeroSteiner** [comment, marked as answer]
If you have one rule and many objects, you'd want to use the [`Rule.filter`](https://zerosteiner.github.io/rule-engine/rule_engine/engine.html#rule_engine.engine.Rule.filter) method. If you have many rules and one object, you'd need to initialize a `Rule` object for each and apply them to your object like: `[rule_text for rule_text in all_rule_texts if Rule(rule_text).matches(object)]`.

**Reply by Nik3181** [reply]
Have many rules and many objects, hence looping theough each one of them is quite slow.

**Reply by tejesh-p** [reply]
Hi @Nik3181. Its late but I am facing a similar issue. did you get any workaround for that scenario?
